src/libint_3center.F

```diff
@@ -313,7 +350,214 @@ SUBROUTINE set_params(lib, ri, rj, rk, li_max, lj_max, lk_max, zeti, zetj, zetk,
+
+      NULLIFY (params, p_work)
+      ALLOCATE (params)
+
+      my_r_cutoff = 0.0_dp
+      IF (PRESENT(r_cutoff)) my_r_cutoff = r_cutoff
+
+      my_omega = 0.0_dp
+      IF (PRESENT(omega)) my_omega = omega
+
+      !Looping over the pgfs
+      DO ipgf = 1, npgfa
+         zeti = zeta(ipgf)
+         a_start = (ipgf-1)*ncoset(la_max)
+
+         DO jpgf = 1, npgfb
+            zetj = zetb(jpgf)
+            b_start = (jpgf-1)*ncoset(lb_max)
+
+            CALL set_params_2c(lib, ra, rb, la_max, lb_max, zeti, zetj, op, &
+                               params_out=params, r_cutoff=my_r_cutoff, omega=my_omega)
+
+            DO li = la_min, la_max
+               a_offset = a_start+ncoset(li-1)
+               ncoa = nco(li)
+               DO lj = lb_min, lb_max
+                  b_offset = b_start+ncoset(lj-1)
+                  ncob = nco(lj)
+
+                  a_mysize(1) = ncoa*ncob
+                  CALL cp_libint_get_2eris(li, lj, lib, p_work, a_mysize)
+
+                  DO j = 1, ncob
+                     p1 = (j-1)*ncoa
+                     DO i = 1, ncoa
+                        p2 = p1+i
+                        int_ab(a_offset+i, b_offset+j) = p_work(p2)
+                     END DO
+                  END DO
+
+               END DO
+            END DO
+
+         END DO
+      END DO
+
+      DEALLOCATE (params)
+
+   END SUBROUTINE
+
+! **************************************************************************************************
+!> \brief Sets the internals of the cp_libint_t object for integrals of type (k|j)
+!> \param lib ..
+!> \param rj ...
+!> \param rk ...
+!> \param lj_max ...
+!> \param lk_max ...
+!> \param zetj ...
+!> \param zetk ...
+!> \param op ...
+!> \param r_cutoff ...
+!> \param omega ...
+!> \param params_in external parameters to use for libint
+!> \param params_out returns the libint parameters computed based on the other arguments
+!> \note The use of params_in and params_out comes from the fact that one might have to swap
+!>       centers 1 and 2 because of angular momenta and pretty much all the parameters of libint
+!>       remain the same upon such a change => might avoid recomputing things over and over again
+! **************************************************************************************************
+   SUBROUTINE set_params_2c(lib, rj, rk, lj_max, lk_max, zetj, zetk, &
+                            op, r_cutoff, omega, params_in, params_out)
+
+      TYPE(cp_libint_t), INTENT(INOUT)                   :: lib
+      REAL(dp), DIMENSION(3), INTENT(IN)                 :: rj, rk
+      INTEGER, INTENT(IN), OPTIONAL                      :: lj_max, lk_max
+      REAL(dp), INTENT(IN), OPTIONAL                     :: zetj, zetk
+      INTEGER, INTENT(IN), OPTIONAL                      :: op
+      REAL(dp), INTENT(IN), OPTIONAL                     :: r_cutoff, omega
+      TYPE(params_2c), OPTIONAL, POINTER                 :: params_in, params_out
```

Comment:
As far as I can see there is no center permutation for the 2-center, does this `params_in` and `params_out` still make sense in this case ?